presto-main/src/main/java/com/facebook/presto/operator/GroupByHash.java

```diff
@@ -64,4 +64,6 @@ static GroupByHash createGroupByHash(
     boolean contains(int position, Page page, int[] hashChannels);
 
     int putIfAbsent(int position, Page page);
+
+    int compare(int leftGroupId, int rightGroupId);
```

Comment:
I can not sort single pages. All positions in all pages produced from `GroupByHash` have to be sorted. So in your approach, I would have to materialize ALL result pages, sort them in memory and then spill them to disk. Problem would be with materialization, which would basically double the memory usage of the operator.

Solutions for that are ugly at best and far from optimal in terms of performance. I don't want to focus on performance with this first iteration, but if there is an easy way to implement something fast, it would be shame to de-optimize code, because later with spilling to highly compressed files stored on SSD should be pretty fast.

Is this `compare` really that  bad?